TIL: how to replace an external Go dependency with a local copy

zmoog opened the pull request:
### Motivation
<!-- Why this pull request? -->

I want to start publishing on my blog a series of small TIL posts and then over time move up to more long form content.

### Change description
<!-- What does your code do? -->

Add a post on how to use `replace` directive to replace a dependency module with locla copy.

### Additional Notes
<!-- Link any useful metadata: Jira task, GitHub issue, ... -->

¯\_(ツ)_/¯

Commit:
Add first draft for the replace go module post

Commit:
Edits to til replace module draft

xrmx commented on content/post/til-replace-module.md:
```diff
@@ -0,0 +1,52 @@
+
+> "Oh, wouldn't be great if the author just added a tiny little log statement in that function?". 
+
+Yeah, it really would, thiw would allow me to trace the value of that variable o try a small change.
```
Thiw -> this

xrmx replied:
O -> or (or to?)

xrmx commented on content/post/til-replace-module.md:
```diff
@@ -0,0 +1,52 @@
+
+That's it!
+
+Now, when you recompile or run your code, the local copy will be used with all your local changes.
```
You should add more emphasis that the change should be transient and never committed

zmoog replied:
Oh, you're right! I didn't think about this aspect, thank you!

Commit:
Fix typos

Commit:
Add a notice for the user

The use of the replace directive for local copies of the modules should not be committed to the repo.

xrmx reviewed (approved):
:tada:

Commit:
Update page metadata

pollon commented on content/post/til-replace-module.md:
````diff
@@ -0,0 +1,59 @@
+)
+```
+
+To replace the module `github.com/zmoog/bar` with a local copy located on your local disk, all you need to do is add the a new line with the [replace](https://golang.org/ref/mod#go-mod-file-replace) directive:
````
all you need to do is add **the** a new line -> all you need to do is add a new line

zmoog replied:
Thank you! ❤️  Fixed in #3.
